pygmt/helpers/utils.py

```diff
@@ -315,6 +315,108 @@ def non_ascii_to_octal(argstr):
     return argstr.translate(str.maketrans(mapping))
 
 
+def build_arg_list(
+    kwdict: dict,
+    confdict: dict | None = None,
+    infile: str | pathlib.PurePath | list[str | pathlib.PurePath] | None = None,
```

Comment:
Either a list or tuple can be passed in, so maybe use [`Sequence`](https://docs.python.org/3/glossary.html#term-sequence)? Or `list | tuple`?
```suggestion
    infile: str | pathlib.PurePath | Sequence[str | pathlib.PurePath] | None = None,
```